src/account/AccountPickScreen.js

```diff
@@ -28,15 +36,21 @@ class AccountPickScreen extends PureComponent<Props> {
   static contextType = TranslationContext;
   context: GetText;
 
-  handleAccountSelect = (index: number) => {
+  handleAccountSelect = async (index: number) => {
+    const _ = this.context;
     const { accounts, dispatch } = this.props;
     const { realm, isLoggedIn } = accounts[index];
     if (isLoggedIn) {
       setTimeout(() => {
         dispatch(accountSwitch(index));
       });
     } else {
-      NavigationService.dispatch(navigateToRealmInputScreen({ realm }));
+      try {
+        const serverSettings: ApiResponseServerSettings = await api.getServerSettings(realm);
+        NavigationService.dispatch(navigateToAuth(serverSettings));
+      } catch (e) {
+        showErrorAlert(_('Failed to connect to realm: {realm}', { realm: realm.toString() }));
```

Comment:
AFAIK we don't want to use "realm" in user-facing strings, switch this to "organization"?

Reply:
Oh, really? I didn't know that; I've been careful to say "realm" for the same reason I'm careful to say "stream" instead of "channel"; I thought it was a product decision. But I guess it could be an outdated and non-user-facing thing, like the way we say "topic" to users even though some legacy parts of the code say "subject". Has there been a discussion on this somewhere, or maybe @alya has more context?

Reply:
I don't know the history, but "organization" is indeed the term we use on all the help pages (e.g. https://zulip.com/help/join-a-zulip-organization) and in the settings menu, so we should stay consistent with that in user-facing strings.

Reply:
Will do, thanks!

Reply:
Yep, and we have this documented in the mobile repo's glossary:
https://github.com/zulip/zulip-mobile/blob/master/docs/glossary.md#realm

Reply:
In this context, probably "server" is better yet: that's the thing we're failing to connect to.